proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java

```diff
@@ -156,33 +160,44 @@ public QueuedAgentService(AgentAPI service, String bufferFile, final int retryTh
         }
       }
-      final ResubmissionTaskQueue taskQueue = new ResubmissionTaskQueue(
-          new FileObjectQueue<>(buffer, new FileObjectQueue.Converter<ResubmissionTask>() {
-            @Override
-            public ResubmissionTask from(byte[] bytes) throws IOException {
-              try {
-                if (bytes.length > 2 && bytes[0] == (byte) 0x1f && bytes[1] == (byte) 0x8b) {
-                  // gzip signature detected (backwards compatibility mode)
-                  Reader reader = new InputStreamReader(new GZIPInputStream(new ByteArrayInputStream(bytes)));
-                  return resubmissionTaskMarshaller.fromJson(reader, ResubmissionTask.class);
-                }
-                ObjectInputStream ois = new ObjectInputStream(new LZ4BlockInputStream(new ByteArrayInputStream(bytes)));
-                return (ResubmissionTask) ois.readObject();
-              } catch (Throwable t) {
-                logger.warning("Failed to read a single retry submission from buffer, ignoring: " + t);
-                return null;
-              }
+      ObjectQueue<ResubmissionTask> queue = new FileObjectQueue<>(buffer,
+          new FileObjectQueue.Converter<ResubmissionTask>() {
+        @Override
+        public ResubmissionTask from(byte[] bytes) throws IOException {
+          try {
+            if (bytes.length > 2 && bytes[0] == (byte) 0x1f && bytes[1] == (byte) 0x8b) {
+              // gzip signature detected (backwards compatibility mode)
+              Reader reader = new InputStreamReader(new GZIPInputStream(new ByteArrayInputStream(bytes)));
+              return resubmissionTaskMarshaller.fromJson(reader, ResubmissionTask.class);
             }
+            ObjectInputStream ois = new ObjectInputStream(new LZ4BlockInputStream(new ByteArrayInputStream(bytes)));
+            return (ResubmissionTask) ois.readObject();
+          } catch (Throwable t) {
+            logger.warning("Failed to read a single retry submission from buffer, ignoring: " + t);
+            return null;
+          }
+        }
 
-            @Override
-            public void toStream(ResubmissionTask o, OutputStream bytes) throws IOException {
-              LZ4BlockOutputStream lz4BlockOutputStream = new LZ4BlockOutputStream(bytes);
-              ObjectOutputStream oos = new ObjectOutputStream(lz4BlockOutputStream);
-              oos.writeObject(o);
-              oos.close();
-              lz4BlockOutputStream.close();
-            }
-          }),
+        @Override
+        public void toStream(ResubmissionTask o, OutputStream bytes) throws IOException {
+          LZ4BlockOutputStream lz4BlockOutputStream = new LZ4BlockOutputStream(bytes);
+          ObjectOutputStream oos = new ObjectOutputStream(lz4BlockOutputStream);
+          oos.writeObject(o);
+          oos.close();
+          lz4BlockOutputStream.close();
+        }
+      });
+
+      try {
+        FileChannel channel = new RandomAccessFile(buffer, "rw").getChannel();
```

Comment:
You also found this trick, please put a comment that this doesn't work on all platforms